### PR TITLE
Build fixes for ASAN

### PR DESCRIPTION
**Pull request by maleadt**
`LDFLAGS` got erroneously populated with `-fsanitize=address`, causing in my case openspecfun to get linked against an incompatible ASAN runtime (because of using `gfortran` instead of `clang`).

I also removed traces of `DEPS_C*FLAGS`, which was only used for the (defunct) building of deps+ASAN.

CI [failure before](https://ci.maleadt.net/buildbot/julia/builders/Julia%20with%20ASAN%20%28x86-64%29/builds/5) (`Your application is linked against incompatible ASan runtimes.`), [after](https://ci.maleadt.net/buildbot/julia/builders/Julia%20with%20ASAN%20(x86-64)/builds/6) (still fails, but due to different issues).

### CODE REVIEW

**Comment by maleadt** [comment]
Trying to avoid another #21624 here -- `DEPS_C.*FLAGS` really seem unused.
I guess this might apply for 0.6 if that window is still open (cc @tkelman)?

**Comment by tkelman** [comment]
DEPS_CFLAGS was added by Keno in #10785. It's possible someone may have been setting it in Make.user or on the make command line, possibly for non sanitizer related purposes, in which case I don't think we need to remove them from a release branch but on master should be okay.

File: deps/tools/common.mk [hunk 1]
```diff
@@ -9,10 +9,10 @@ ifneq ($(USEMSVC), 1)
 CONFIGURE_COMMON += LDFLAGS="$(LDFLAGS) -Wl,--stack,8388608"
 endif
 endif
-CONFIGURE_COMMON += F77="$(FC)" CC="$(CC) $(DEPS_CFLAGS)" CXX="$(CXX) $(DEPS_CXXFLAGS)"
+CONFIGURE_COMMON += F77="$(FC)" CC="$(CC)" CXX="$(CXX)"
 
-CMAKE_CC_ARG := $(CC_ARG) $(DEPS_CFLAGS)
-CMAKE_CXX_ARG := $(CXX_ARG) $(DEPS_CXXFLAGS)
+CMAKE_CC_ARG := $(CC_ARG)
+CMAKE_CXX_ARG := $(CXX_ARG) $
```
Comment (tkelman) [inline deps/tools/common.mk]: leftover $

**Comment by maleadt** [comment]
> I don't think we need to remove them from a release branch but on master should be okay.

Sounds reasonable. Do I create a second PR for backporting the core fix, or just leave it as-is?
ASAN is part of the devdocs, so it might make sense to fix it for the release.

**Comment by tkelman** [comment]
I can get that commit in a backport batch

**Pull request merged**